### PR TITLE
task solution

### PR DESCRIPTION
**Pull request by vladyslav-harkusha**
[DEMO LINK](https://vladyslav-harkusha.github.io/react_movies-list-add-form/)

### CODE REVIEW

**Review by Yuriy-Komolov — APPROVED** [review]

**Comment by Yuriy-Komolov** [comment]
Good Job!

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
-export const App = () => {
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={addMovie} />
```
Comment (sergii-nosachenko) [inline src/App.tsx:22]:
```suggestion
        <NewMovie onAdd={addMovie} key={count} />
```
Move counter logic here. When you'll increase count your component will be created from scratch with default values and no errors.

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,87 @@
...
+      imdbId,
+    });
+
+    setTitle('');
+    setDesription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+
+    setCount(current => current + 1);
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:35]:
```suggestion
```
Move setCount to parent component and use it in onAdd handler. You don't need to reset your form.

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,87 @@
...
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={value => setTitle(value)}
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:52]:
```suggestion
        onChange={setTitle}
```
There is no event here. TextField component calls your handler with field value, so you don't need wrapper here.

Fix it everywhere.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]
You need to add additional check if your fields consists of spaces only.
Screenshot 2022-10-31 at 17 15 53

**Review by sergii-nosachenko — APPROVED** [review]
Perfect 🔥

**Pull request closed without merging**